osbuild/pipeline.py

```diff
@@ -282,20 +282,14 @@ def build_stages(self, object_store, interactive, libdir, secrets):
             if stage.checkpoint:
                 object_store.commit(tree, stage.id)
 
-        results["tree_id"] = self.tree_id
         return results, build_tree, tree
 
-    def assemble(self, object_store, build_tree, tree, interactive, libdir):
+    def assemble(self, object_store, build_tree, tree, interactive, libdir, output_directory=None):
```

Comment:
I think `output_directory` would be fine without a default argument because the (internal) callers always pass it anyway. Shouldn't block this going in though.

Reply:
I have a follow-up for that. Thanks!

Reply:
Ahhhh, I dropped the fixup again for now. Thing is, the output-directory is optional, and we have to keep the conditional `if output_directory: ...`. I think it is easier to notice that the value can reasonable be `None` with this default-value set.

What do you think? Is it fine to not have the default value but still check for it being `None` in all uses?

Reply:
Makes sense! We could do `output_directory: Optional[str]` or something but I think having the default value as an indicator is also good!